src/hooks/queries/useAuthService.ts

```diff
@@ -0,0 +1,48 @@
+import { useMutation } from '@tanstack/react-query';
+import { AxiosError } from 'axios';
+import { LoginResponse } from '@/models/auth/authApiModels';
+import { useAuthStore } from '@/stores/authStore';
+import { useModalStore } from '@/stores/modalStore';
+import type { LoginForm } from '@/models/auth/authModels';
+import { login } from '../../services/api/authApi';
+import { useUserStore } from '@/stores/userStore';
+import { userData } from '@/services/api/userApi';
+
+export const useLogin = () => {
+  const { closeModal } = useModalStore();
+  const setUser = useUserStore((state) => state.setUser);
+
+  return useMutation<LoginResponse, AxiosError, LoginForm>({
+    mutationFn: login,
+    onSuccess: async (response) => {
+      const { accessToken, refreshToken } = response.data;
+
+      // 토큰을 전역 저장소에 저장 (zustand에서 persist로 localStorage에 저장)
+      useAuthStore.getState().login(accessToken, refreshToken);
+
+      try {
+        // 로그인 후 유저 데이터 가져오기
+        const userDataResponse = await userData();
+        const data = userDataResponse.data;
+
+        // 유저 데이터를 Zustand 스토어에 저장
+        setUser({
+          name: data.username,
+          email: data.email,
+          roles: data.interestJobs,
+          skills: data.skills, // 백엔드에서 아직 안넘겨줌. 빈값으로 설정
+        });
+
+        closeModal();
+        alert('로그인에 성공했습니다.');
+      } catch (error) {
+        console.error('유저 데이터 가져오기 실패:', error);
+        alert('로그인은 성공했지만 유저 데이터를 가져오는데 실패했습니다.');
+      }
+    },
+    onError: (error) => {
+      alert(error);
```

Comment:
추후 alert 다 메세지 박스로 바꿔야 하는데 저도 일단은 미뤄둔 상태입니다....
그래도 콘솔처럼 어떤 에러인지 알 수 있게 처리만 부탁드립니다. 
```suggestion
      alert('로그인에 실패했습니다. 다시 시도해 주세요.');
```

Reply:
저희 오류 로그 남기는 규칙을 하나 정해야할 것 같다고 생각하긴 했습니다!
axios 호출에도 남기고, query hook에서도 남기면 좀 불필요할 것 같아.. 규칙을 정합시다!

Reply:
✅ be677dccef09776d5a85f0a6c5075f3856bd58ec